### PR TITLE
neomutt: 20231221 -> 20240201

### PR DESCRIPTION
**Pull request by Ma27**
## Description of changes
ChangeLog: https://github.com/neomutt/neomutt/releases/tag/20240201

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by afh** [comment]
Result of `nixpkgs-review pr 285578` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>neomutt</li>
  </ul>
</details>

File: pkgs/applications/networking/mailreaders/neomutt/default.nix [hunk 1]
```diff
@@ -6,14 +6,14 @@
...
-  version = "20231221";
+  version = "20240201";
   pname = "neomutt";
 
   src = fetchFromGitHub {
     owner  = "neomutt";
     repo   = "neomutt";
     rev    = version;
-    sha256 = "sha256-IXly2N/DD2+XBXVIXJw1sE/0eJwbUaONDNRMi7n1T44=";
+    sha256 = "sha256-1yOjS2uHY66cKhUB9nwnRUpfg9VK5uWU9TcTXbX7shU=";
```
Comment (afh) [inline pkgs/applications/networking/mailreaders/neomutt/default.nix:16]: Minor nit:

```suggestion
    hash = "sha256-1yOjS2uHY66cKhUB9nwnRUpfg9VK5uWU9TcTXbX7shU=";
```

ℹ️ The same change could be made for the `test-file` and `contrib` [below](https://github.com/NixOS/nixpkgs/pull/285578/files#diff-f6b0ff192b29c6d77d458a8d627b07d98bccb5088b959211e883cabc8da788e9R88-R101).

**Review by afh — APPROVED** [review]
Thanks for the update, @Ma27.

If you feel like it maybe you'd like to seize the opportunity to improve the default.nix by using `hash` instead of `sha256` in `fetchFromGitHub` (see suggestion and comment below).

**Comment by afh** [comment]
@Ma27 [neomutt 20240323](https://github.com/neomutt/neomutt/releases/tag/20240323) has been released. Do you have time and interest in updating this PR?

**Comment by alyssais** [comment]
["DO NOT USE release 2024-03-23"](https://mailman.neomutt.org/pipermail/neomutt-devel-neomutt.org/2024-March/000916.html)

**Comment by afh** [comment]
Yikes! I only saw the release notification in my RSS reader, I'll make it a habit to check the release notes before posting comments like my previous one. Thanks, @alyssais, for providing the important info about that release :pray:

**Comment by alyssais** [comment]
That information was not available at the time you posted. :) It was released and then retracted later.

**Comment by afh** [comment]
@Ma27 [neomutt 20240416](https://github.com/neomutt/neomutt/releases/tag/20240416) has been released. Do you have time and interest in updating this PR, provided the release isn't revoked?

**Comment by Ma27** [comment]
Going to close this now.
Currently evaluating aerc to see if it fits better, so I won't be spending much time on testing neomutt anyways currently, sorry.

Feel free to file a new PR :)

**Pull request closed without merging**